modules/core/src/lib/attribute-manager.js

```diff
@@ -392,4 +425,12 @@ export default class AttributeManager {
       });
     }
   }
+
+  _getChangedShaderAttributes(shaderAttributes, attributeTransitionManager, changedAttributes) {
+    for (const shaderAttributeName in shaderAttributes) {
+      if (!attributeTransitionManager.hasAttribute(shaderAttributeName)) {
+        changedAttributes[shaderAttributeName] = shaderAttributes[shaderAttributeName];
```

Comment:
We usually avoid changing arguments, instead I would return a new object with changed attributes.

There will be some overhead when merging objects. However attributes shouldn't change that frequently and any overhead should be dwarfed by attribute creation costs.